AngularApp/src/app/shared/services/resource.service.ts

```diff
@@ -1,26 +1,52 @@
 import { Injectable } from '@angular/core';
-import { ActivatedResource, ResourceType } from '../models/resources';
 import { Observable } from 'rxjs/Observable';
+import { ArmResource } from '../models/resources';
 
 @Injectable()
-export abstract class ResourceService {
+export class ResourceService {
+
+  public lensPrefix: string;
+  public imgSrc: string;
+  public versionPrefix: string;
+  public templateFileName: string;
+
+  protected _requestBody: any = null;
+  protected _armResource: ArmResource;
+  protected _initialized: Observable<boolean>;
 
   constructor() { }
 
-  public abstract lensPrefix: string;
-  public abstract imgSrc: string;
+  initialize(armResource: ArmResource, templateFileName: string = '', lensPrefix: string = 'App', imgSrc: string = '', versionPrefix: string = '') {
```

Comment:
Is the lensPrefix used in UI?
It might not be bad idea to just use Applens (to spread the brand..)

Reply:
agree